packages/surveys/src/lib/utils.ts

```diff
@@ -69,3 +69,23 @@ export const calculateElementIdx = (survey: TSurvey, currentQustionIdx: number):
   if (possibleNextQuestions.includes("end")) elementIdx = middleIdx;
   return elementIdx;
 };
+
+export function calculateProgress(questionId: string, survey: TSurvey, progress: number) {
```

Comment:
What was the idea behind moving the code away from the ProgressBar component?
As the the `calculateProgress` function was only used for this component I liked this approach.
I think now it's harder to follow the logic of the progressBar and also adds more complexity to the Survey.tsx file which is already too chaotic ;-)

Reply:
Yes that is correct, moved the code to ProgressBar.tsx again 😊